src/vite/plugin.tsx

```diff
@@ -91,13 +96,54 @@ export const remixDevTools: (args?:RemixViteConfig) => Plugin[] = (args) => {
           }))
         });
 
-        if(!server.config.isProduction){
-           const { exec } = await import ("node:child_process") 
-          server.hot.on("open-source", (data) => { 
-          const source = data.data.source; 
-          const line = data.data.line;
-          exec(`code -g "${normalizePath(source)}:${line}"`)
-          })
+        if (!server.config.isProduction) {
+          const { exec } = await import("node:child_process");
+
+          server.hot.on("open-source", ({ data }: OpenSourceData) => {
+            const { source, line, routeID } = data;
+            const lineNum = line ? `:${line}` : "";
+
+            if (source) {
+              exec(`code -g "${normalizePath(source)}${lineNum}"`);
+              return;
+            }
+
+            if (!source && routeID) {
```

Comment:
the first part of the check is redundant as you already checked if the source exists above, so this can be if(routeID)

Reply:
what happens if the routeID isn't there? I would expect it to always be sent, or?